qiskit_experiments/library/characterization/t2hahn.py

```diff
@@ -0,0 +1,157 @@
+        """Default experiment options.
+
+        Experiment Options:
+            delays (Iterable[float]): Delay times of the experiments.
+            unit (str): Unit of the delay times. Supported units are
+                's', 'ms', 'us', 'ns', 'ps', 'dt'.
+        """
+        options = super()._default_experiment_options()
+
+        options.delays = None
+        options.unit = "s"
+
+        return options
+
+    def __init__(
+        self,
+        qubit: Union[int, Iterable[int]],
+        delays: Union[List[float], np.array],
+        unit: str = "s",
+    ):
+        """
+        **T2 - Hahn Echo class**
+        Initialize the T2 - Hahn Echo class
+        Args:
+            qubit: the qubit under test.
+            delays (List[float)): delay times of the experiments.
+            unit: Optional, time unit of `delays`.
+                Supported units: 's', 'ms', 'us', 'ns', 'ps', 'dt'. The unit is
+                used for both T2Ramsey and for the frequency.
+
+         Raises:
+             QiskitError : Error for invalid input.
+        """
+        # Initialize base experiment
+        super().__init__([qubit])
+        # Set configurable options
+        self.set_experiment_options(delays=delays, unit=unit)
+        self._verify_parameters()
+
+    def _verify_parameters(self):
+        """
+        Verify input correctness, raise QiskitError if needed.
+        Args:
+            qubit: the qubit under test.
+
+        Raises:
+            QiskitError : Error for invalid input.
+        """
+        if any(delay <= 0 for delay in self.experiment_options.delays):
+            raise QiskitError(
+                f"The lengths list {self.experiment_options.delays} should only contain "
+                "positive elements."
+            )
+        if len(set(self.experiment_options.delays)) != len(self.experiment_options.delays):
+            raise QiskitError(
+                f"The lengths list {self.experiment_options.delays} should not contain "
+                "duplicate elements."
+            )
+
+        if any(
+            self.experiment_options.delays[idx - 1] >= self.experiment_options.delays[idx]
+            for idx in range(1, len(self.experiment_options.delays))
+        ):
+            raise QiskitError(
+                f"The number of identity gates {self.experiment_options.delays} should "
+                "be increasing."
+            )
+
+    def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+        """
+        Args:
+            backend: Optional, a backend object.
+
+        Returns:
+            The experiment circuits.
+
+        """
+
+        circuits = []
+        qubit = list(self._physical_qubits)
+        for circ_index, delay in enumerate(self.experiment_options.delays):
+            circ = QuantumCircuit(1, 1)
+            # First Y rotation in 90 degrees
+            circ.ry(np.pi / 2, qubit)  # Bring to qubits to X Axis
```

Comment:
All the gates are to be applied on qubit 0, which is the only qubit in the circuit. Subsequent code in BaseExperiment maps it to the correct qubit that was given in qubit.

Reply:
Changed that all the gates will work on qubit '0' exclusively.